feat: cross-machine usage aggregation + automatic sync

Yeachan-Heo opened the pull request:
## Summary
- **Cross-machine aggregation**: Usage from multiple devices is now properly SUMMED instead of overwritten
- **Automatic sync**: New `tokscale sync` commands for hourly crontab automation
- Closes #52 

## Problem Solved
- Before: Machine A=1000 tokens, Machine B=500 tokens → Result: 500 (Machine A data LOST)
- After: Machine A=1000 tokens, Machine B=500 tokens → Result: 1500 (properly aggregated)

## New Commands
```bash
tokscale sync setup    # Set up hourly cron job
tokscale sync status   # Check if sync is active
tokscale sync remove   # Remove the cron job
```

## Changes
- Track per-device contributions in `devices` field
- Added `--quiet` flag to `tokscale submit` for cron compatibility
- Sync logs to `~/.config/tokscale/sync.log`
- Legacy data migrated to `devices.__legacy__`

## Testing
- 28 tests covering: cross-device aggregation, same-device replacement, legacy migration, multi-model scenarios

Yeachan-Heo committed:
feat(cli): add --quiet flag to submit for cron compatibility

Yeachan-Heo committed:
fix(frontend): track per-device contributions for proper cross-machine aggregation

- Add DeviceSourceData interface for per-device contribution tracking
- Add devices field to SourceBreakdownData type in helpers.ts and schema.ts
- Modify mergeSourceBreakdowns() to accept deviceId parameter
- Add recalculateSourceAggregate() helper to sum across devices
- Migrate existing data without devices field to __legacy__ device
- Update submit route to pass tokenRecord.tokenId as deviceId
- Add tests for same-device replacement, cross-device aggregation, and legacy migration

Yeachan-Heo committed:
feat(cli): add tokscale sync command for automatic hourly submissions

- Add sync.ts with setupSync(), removeSync(), syncStatus() functions
- Support crontab (macOS/Linux) and Task Scheduler (Windows)
- Use process.argv[1] for CLI path resolution
- Crontab entry uses --quiet flag for silent execution
- Log output to ~/.config/tokscale/sync.log

Yeachan-Heo committed:
docs: add automatic sync and cross-machine aggregation documentation

Yeachan-Heo committed:
fix: address Oracle review - prevent double-count, shell injection, cron cleanup

- Store devices[deviceId] on new day inserts to prevent double-counting
  when same device resubmits (was migrating to __legacy__ and adding)
- Fix shell injection risk in crontab setup using printf with escaped quotes
- Fix Windows Task Scheduler logging by wrapping in cmd.exe for redirections
- Use specific grep pattern 'tokscale submit --quiet' to avoid removing
  unrelated cron jobs
- Add tests for insert→resubmit flow to verify no double-counting

Yeachan-Heo committed:
fix: address Oracle review round 2 - defensive defaults, cron marker, direct tests

- Add ?? {} defensive defaults for models access in mergeSourceBreakdowns()
- Use TOKSCALE_SYNC_MANAGED marker for cron entries (prevents false matches)
- Windows sync now uses .cmd script file approach (simpler, no quoting issues)
- Add direct mergeSourceBreakdowns() function call tests for legacy data handling

Yeachan-Heo committed:
fix: address Oracle review round 3 - legacy inherits to current device, Windows cmd wrapper

Yeachan-Heo committed:
fix: preserve other devices contributions in merge, proper legacy migration

Yeachan-Heo committed:
fix: restore __legacy__ migration to preserve historical data, add type coercion

- Legacy migration now uses '__legacy__' device instead of current deviceId
- This preserves historical data separately from new device contributions
- Added Number(...) || 0 for all arithmetic in recalculateSourceAggregate
- Handles potential string values from JSON serialization
- Updated tests to expect __legacy__ entry for legacy data migrations

Yeachan-Heo committed:
fix: handle empty devices aggregation, complete type coercion

- Reset aggregates before empty devices check to prevent stale values
- Use Number() coercion in recalculateDayTotals for consistency

Yeachan-Heo committed:
fix: complete type coercion in route.ts aggregation

Convert all || 0 patterns to Number(...) || 0 to handle cases where
values may be strings from JSON parsing or database retrieval.

junhoyeo commented:
Hey @Yeachan-Heo, thanks for opening a PR tackling two tasks. Appreciate the contribution despite the complexity! 🙌
I'll review and merge once everything looks good.

One heads-up: the commits and PR body had the `"Co-authored-by: Junho Yeo <...>"` footer attached for some reason, so I removed those and force-pushed. Just wanted to let you know!

junhoyeo committed:
fix: address security and stability issues in cross-machine sync

- Windows sync: Mark as experimental/disabled pending security review
- bunx detection: Prevent sync setup from temp cache paths with clear error message
- Crontab security: Add path validation to prevent injection via control chars
- Legacy migration: Fix modelId→models conversion to preserve model breakdown

Closes issues identified in PR #55 review.

junhoyeo commented:
@Yeachan-Heo fyi, I will hold this pr from merging before #64 -- there will be significant changes there!

junhoyeo commented:
> Legacy data with `modelId` only → creates correct model breakdown

As of today, no users have the deprecated `modeld` field!

junhoyeo committed:
chore: merge main into PR branch, resolve conflicts

Resolved conflicts:
- README.md: Keep main's structure (Overview before Contents), add Automatic Sync to TOC
- cli.ts: Combine knownCommands to include both 'sync' and 'pricing'
- submit.ts: Keep quiet flag feature with main's simplified structure

junhoyeo committed:
refactor: remove --quiet flag from submit command

The submit command already has minimal necessary logs, and output is
redirected to log file for cron jobs anyway. --no-spinner exists for
scripts that need clean stdout.

jakublala commented:
Hey @Yeachan-Heo and @junhoyeo  — I really want this feature and dug into why it's stuck: the CLI portion (`cli.ts` + `sync.ts`) needs porting to Rust under `crates/tokscale-cli/` after #64's rewrite, and the three frontend files conflict mostly from main's global `source → client` rename plus `route.ts`'s batched `toUpdate`/`toInsert` refactor — the device-aggregation logic itself is clean and rebasable, and the Rust port reuses existing crates (`clap`, `dirs`, `std::process::Command`) with no new deps. If you're swamped, I'd be glad to do the Rust port and the rebase as a follow-up PR with you as co-author, or prep a patch you can pull straight onto `feat/cross-machine-aggregation-sync`. Whichever you prefer — just want to help get this over the line!

junhoyeo commented:
Hey @Yeachan-Heo — thanks again for this one, and sorry it sat so long. Closing as superseded, with full context:

**The core problem this PR tackled (multi-machine submits overwriting each other) has since been fixed on main**, via a different design that landed across #517 (device-aware usage merges), #593 (per-device read + rename APIs), and #609 (legacy bucket adoption):

- The CLI now persists a UUID device id (`crates/tokscale-cli/src/device.rs`, stored in `~/.config/tokscale/device.json`, overridable via `TOKSCALE_DEVICE_ID`/`TOKSCALE_DEVICE_NAME`) and attaches it to every submit payload.
- The server stores per-device contributions **relationally** — a `submitted_devices` table unique on `(user_id, device_key)`, with `daily_breakdown` rows scoped by `submitted_device_id` — instead of a nested `devices` map inside the JSONB breakdown. Totals are recomputed by summing across all devices, so machine A + machine B = A + B, and a resubmit from one machine only replaces that machine's slice.
- Pre-device-era history is adopted into a `legacy-default` bucket on the first device-aware submit, so nothing is double-counted.

The CLI half of this PR also lost its home: `packages/cli` was replaced by the Rust CLI (`crates/tokscale-cli`), so the TypeScript `sync.ts`/crontab code no longer has a target.

**The auto-sync idea is still genuinely good and still missing** — there's no scheduler in the Rust CLI today. I'm tracking it separately as a Rust CLI feature with credit to this PR (including the hardening requirements from the January review round: locking against concurrent runs and safe cron entry management). You'd be very welcome to take that one. 🙏
